cli/loader_lib.c

```diff
@@ -163,6 +240,32 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
     // Introspect to find our own path
     const char * lib_dir = jl_get_libdir();
 
+#ifdef LIBSTDCXXPROBE
+    void* handle;
+    int do_probe = 1;
+    char* probevar = getenv("JLLIBSTDCXXPROBE");
+    if (probevar) {
+        if (strcmp(probevar, "1") == 0 || strcmp(probevar, "yes"))
+            do_probe = 1;
+        else if (strcmp(probevar, "0") == 0 || strcmp(probevar, "no"))
+            do_probe = 0;
+    }
+    if (do_probe) {
+        char cxxpath[JL_PATH_MAX];
+        size_t pathlen = 0;
+        if (!libstdcxxprobe(cxxpath, &pathlen)) {
+            jl_loader_print_stderr("ERROR: Syscalls failed while finding which libstdcxx path to load!\n");
+            exit(1);
```

Comment:
It might be better to make each error site explicitly fail, or not to fail at all, than to have this cryptic error message

Reply:
Added error handling to every site that could possibly fail (except inside the reporting of other failures before an exit, but if fprintf errors I think we're doomed anyway)